tools/git-hooks/pre-commit

```diff
@@ -43,7 +48,7 @@ fi
 cpp_files=$(git diff --cached --name-only --diff-filter=ACMRT | grep -E "\.(c|cpp|inl|h|hpp)$")
 if [[ -n ${cpp_files} ]]; then
 
-    CLANG_CMD="clang-format-10"
+    CLANG_CMD="clang-format-12"
```

Comment:
This is very ubuntu/debian specific. The clang package which is provided by https://clang.llvm.org comes with `clang-format` only. I would recommend the following alternative.

```sh
## when it comes first clang-format-12 has precedence if it exists
if [[ ! -z $(command -v clang-format-12 2>/dev/null) ]]; then 
    CLANG_CMD=$(command -v clang-format-12)
elif [[ ! -z $(command -v clang-format 2>/dev/null) ]]; then
    CLANG_CMD=$(command -v clang-format)
else
    echo clang-format is not installed
    exit
fi
```